[pytorch] rename BUILD_ATEN_MOBILE to INTERN_BUILD_MOBILE and make it private

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19762 [pytorch] CMakeLists changes to enable libtorch for Android
* **#19942 [pytorch] rename BUILD_ATEN_MOBILE to INTERN_BUILD_MOBILE and make it private**
* #19761 [pytorch] add new macro TORCH_MOBILE for libtorch mobile build
* #19760 [pytorch] fix labs warning in THTensorMoreMath.cpp

Summary:
Currently this macro is automatically set when it's building for
Android/iOS - doesn't seem to be necessary to expose this macro
externally. Also remove 'ATEN' from the name as we are going to add
dedicated macro to control Aten build.

Test Plan:
- Build caffe2 mobile library;
- Build libtorch mobile library with stacked diff;
- Build pytorch;

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D15144325](https://our.internmc.facebook.com/intern/diff/D15144325)

ljk53 committed:
[pytorch] rename BUILD_ATEN_MOBILE to INTERN_BUILD_MOBILE and make it private

Summary:
Currently this macro is automatically set when it's building for
Android/iOS - doesn't seem to be necessary to expose this macro
externally. Also remove 'ATEN' from the name as we are going to add
dedicated macro to control Aten build.

Test Plan:
- Build caffe2 mobile library;
- Build libtorch mobile library with stacked diff;
- Build pytorch;

Reviewers:

Subscribers:

Tasks:

Tags:

dzhulgakov reviewed (commented):
Can we add a lengthy comment somewhere (in root CMakeLists.txt or in setup.py, probably the former) describing how different flags are setup and how they interact?

ljk53 committed:
Update on "[pytorch] rename BUILD_ATEN_MOBILE to INTERN_BUILD_MOBILE and make it private"

[pytorch] rename BUILD_ATEN_MOBILE to INTERN_BUILD_MOBILE and make it private

Summary:
Currently this macro is automatically set when it's building for
Android/iOS - doesn't seem to be necessary to expose this macro
externally. Also remove 'ATEN' from the name as we are going to add
dedicated macro to control Aten build.

Test Plan:
- Build caffe2 mobile library;
- Build libtorch mobile library with stacked diff;
- Build pytorch;

Reviewers:

Subscribers:

Tasks:

Tags:

gh-metadata: pytorch pytorch 19942 gh/ljk53/15/head

ljk53 commented:
@dzhulgakov root CMakeLists.txt is probably better for the sake of maintainability.
This PR only renames BUILD_ATEN_MOBILE to better reflect its true semantic - I just commented it. Will incrementally comment other touched build macros in subsequent PRs.
